project_task_knowledge/views/project_task_views.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="action_project_task_document_page" model="ir.actions.act_window">
```

Comment:
You can refactor the action definition like OCA standard does (https://github.com/OCA/knowledge/blob/14.0/document_page_project/views/document_page_views.xml#L13-L35) and achieve this behavior (with folder browser), I'm not sure... but could you try?

![imagen](https://user-images.githubusercontent.com/12749832/223950937-0ffa820a-dafc-4be9-b1cb-bb320608d784.png)